jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java

```diff
@@ -31,29 +36,122 @@
  */
 public enum HttpCompliance 
 { 
-    LEGACY(EnumSet.noneOf(HttpRFC.class)), 
-    RFC2616(EnumSet.complementOf(EnumSet.of(
-            HttpRFC.RFC7230_3_2_4_WS_AFTER_FIELD_NAME,
-            HttpRFC.RFC7230_3_2_4_NO_FOLDING,
-            HttpRFC.RFC7230_A2_NO_HTTP_9))), 
-    RFC7230(EnumSet.allOf(HttpRFC.class)), 
-    ;
+    // TODO in Jetty-10 convert this enum to a class so that extra custom modes can be defined dynamically
+    LEGACY(sectionsBySpec("LEGACY")), 
+    RFC2616(sectionsBySpec("RFC2616")), 
+    RFC7230(sectionsBySpec("RFC7230,-RFC7230_3_1_1_METHOD_CASE_SENSITIVE")), // TODO fix in Jetty-10
+    CUSTOM0(sectionsByProperty("CUSTOM0")),
```

Comment:
Why are these here?
Why can't someone just create a HttpCompliance object instead? (change from enum to class with static definitions)

Reply:
Because currently HttpCompliance is an enum.  To make it dynamic so that anyone can add new modes, would require it to be a class.   This can be done in Jetty-10, but it is too much of a breaking change for 9.4.x as it would break compiled code.

I do have a TODO in there about how it should be converted to a class in jetty-10